### PR TITLE
Fix cmake warnings, have CI test for CMake warnings

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Fix all of our CMake warnings other than the python bindings ones. There are a lot there and I didn't introduce that code.
- Have CI test for CMake warnings from uninitialized variables and fail if there are any. We've had a few bugs in our CMake code because we weren't testing for warnings and there.

### Types of changes:

- [x] Bugfix
- [x] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [x] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by nilsdeppe** [comment]
Rebased on develop

**Review by kidder — DISMISSED** [review]

**Comment by kidder** [comment]
ping @nilsleiffischer

File: cmake/FindBlaze.cmake [hunk 1]
```diff
@@ -11,30 +17,32 @@ find_path(
...
-
-set(BLAZE_VERSION
-  "${BLAZE_MAJOR_VERSION}.${BLAZE_MINOR_VERSION}"
-  )
+set(BLAZE_VERSION "")
 
-set(Blaze_VERSION
-  "${BLAZE_MAJOR_VERSION}.${BLAZE_MINOR_VERSION}"
-  )
+if(EXISTS "${BLAZE_INCLUDE_DIRS}/blaze/system/Version.h")
```
Comment (nilsvu) [inline cmake/FindBlaze.cmake:22]: Should a warning be printed when this check fails?

File: cmake/FindLIBXSMM.cmake [hunk 2]
```diff
@@ -5,42 +5,52 @@
...
+  set(LIBXSMM_ROOT "")
+  set(LIBXSMM_ROOT $ENV{LIBXSMM_ROOT})
+endif()
+
 # find the LIBXSMM include directory
 find_path(LIBXSMM_INCLUDE_DIRS libxsmm.h
-    PATH_SUFFIXES include
-    HINTS ${LIBXSMM_ROOT} ENV LIBXSMM_ROOT)
+  PATH_SUFFIXES include
+  HINTS ${LIBXSMM_ROOT} ENV LIBXSMM_ROOT)
```
Comment (nilsvu) [inline cmake/FindLIBXSMM.cmake]: These `ENV` hints are not needed anymore if you pull it out of `ENV` above. I leave it up to you if you want to remove them

File: cmake/FindYapf.cmake [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  # Need to set to empty to avoid warnings with --warn-uninitialized
+  set(YAPF_ROOT "")
+  set(YAPF_ROOT $ENV{YAPF_ROOT})
+endif()
+
+# Look for an executable called yapf
+find_program(
+  YAPF_EXECUTABLE
+  NAMES yapf
+  HINTS${YAPF_ROOT}
```
Comment (nilsvu) [inline cmake/FindYapf.cmake]: space missing?

File: cmake/FindYapf.cmake [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  DOC "Path to yapf executable")
+
+execute_process(COMMAND "${YAPF_EXECUTABLE}" "--version"
+  RESULT_VARIABLE RESULT
+  OUTPUT_VARIABLE OUTPUT
```
Comment (nilsvu) [inline cmake/FindYapf.cmake]: `VERSION_RESULT` and `VERSION_OUTPUT`?

File: cmake/FindYapf.cmake [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+include(FindPackageHandleStandardArgs)
+
+if(RESULT MATCHES 0)
+  string(REGEX MATCH "[0123456789]+\.[0123456789]+[\.]?[0123456789]*"
```
Comment (nilsvu) [inline cmake/FindYapf.cmake]: `[0-9]`

**Comment by nilsdeppe** [comment]
Rebased on develop and pushed fixups. Thanks for the review @nilsleiffischer!

**Review by nilsvu — COMMENTED** [review]
LGTM please squash

**Comment by nilsdeppe** [comment]
Okay, squashed. Thanks for the reviews @nilsleiffischer and @kidder!

**Review by nilsvu — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**